File: cielo24_PHP/tests/AccessTest.php
<?php

require_once("ActionsTest.php");

class AccessTest extends ActionsTest
{

    public function testLoginPasswordNoHeaders()
    {
        // Username, password, no headers
        $this->apiToken = $this->actions->login($this->config->username, $this->config->password, null, false);
        $this->assertEquals(32, strlen($this->apiToken));
    }

    public function testLoginPasswordHeaders()
    {
        // Username, password, headers
        $this->apiToken = $this->actions->login($this->config->username, $this->config->password, null, true);
        $this->assertEquals(32, strlen($this->apiToken));
    }

    public function testLoginSecureKeyNoHeaders()
    {
        // Username, secure key, no headers
        $this->apiToken = $this->actions->login($this->config->username, null, $this->secureKey, false);
        $this->assertEquals(32, strlen($this->apiToken));
    }

    public function testLoginSecureKeyHeaders()
    {
        // Username, secure key, headers
        $this->apiToken = $this->actions->login($this->config->username, null, $this->secureKey, true);
        $this->assertEquals(32, strlen($this->apiToken));
    }

    public function testLogout()
    {
        // Logout
        $this->actions->logout($this->apiToken);
        $this->apiToken = null;
    }

    public function testGenerateApiKeyForceNew()
    {
        $this->secureKey = $this->actions->generateAPIKey($this->apiToken, $this->config->username, true);
        $this->assertEquals(32, strlen($this->secureKey));
    }

    public function testGenerateApiKeyNotForceNew()
    {
        $this->secureKey = $this->actions->generateAPIKey($this->apiToken, $this->config->username, false);
        $this->assertEquals(32, strlen($this->secureKey));
    }

    public function testRemoveApiKey()
    {
        $this->actions->removeAPIKey($this->apiToken, $this->secureKey);
        $this->secureKey = null;
    }

    public function testUpdatePassword()
    {
        $this->actions->UpdatePassword($this->apiToken, $this->config->newPassword);
        $this->actions->UpdatePassword($this->apiToken, $this->config->password);
    }
}

File: cielo24_Java/cielo24/src/test/AccessTest.java
import static org.junit.Assert.*;

import java.io.IOException;

import org.junit.Test;

import cielo24.utils.WebException;

public class AccessTest extends ActionsTest {
    
    @Test
    public void testLoginPasswordNoHeaders() throws IOException, WebException {
        // Username, password, no headers
        this.apiToken = this.actions.login(this.config.username, this.config.password, false);
        assertEquals(32, this.apiToken.toString().length());
    }

    @Test
    public void testLoginPasswordHeaders() throws IOException, WebException {
        // Username, password, headers
        this.apiToken = this.actions.login(this.config.username, this.config.password, true);
        assertEquals(32, apiToken.toString().length());
    }

    @Test
    public void testLoginSecureKeyNoHeaders() throws IOException, WebException {
        // Username, secure key, no headers
        this.apiToken = this.actions.login(this.config.username, this.secureKey, false);
        assertEquals(32, this.apiToken.toString().length());
    }

    @Test
    public void testLoginSecureKeyHeaders() throws IOException, WebException {
        // Username, secure key, headers
        this.apiToken = this.actions.login(this.config.username, this.secureKey, true);
        assertEquals(32, apiToken.toString().length());
    }

    @Test
    public void testLogout() throws IOException, WebException {
        // Logout
        this.actions.logout(this.apiToken);
        this.apiToken = null;
    }

    @Test
    public void testGenerateApiKey() throws IOException, WebException {
        this.secureKey = this.actions.generateAPIKey(this.apiToken, this.config.username, false);
        assertEquals(32, secureKey.toString().length());
    }

    @Test
    public void testGenerateApiKeyForceNew() throws IOException, WebException {
        this.secureKey = this.actions.generateAPIKey(this.apiToken, this.config.username, true);
        assertEquals(32, this.secureKey.toString().length());
        this.actions.removeAPIKey(this.apiToken, this.secureKey);
    }

    @Test
    public void testRemoveApiKey() throws IOException, WebException {
        this.actions.removeAPIKey(this.apiToken, this.secureKey);
        this.secureKey = null;
    }

    @Test
    public void testUpdatePassword() throws IOException, WebException {
        this.actions.updatePassword(this.apiToken, this.config.newPassword);
        this.actions.updatePassword(this.apiToken, this.config.password);
    }
}